PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/console/ServerlessSparkScalaLivyConsoleRunConfiguration.kt

```diff
@@ -0,0 +1,71 @@
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.azure.hdinsight.spark.console
+
+import com.intellij.execution.Executor
+import com.intellij.execution.configurations.RunProfileState
+import com.intellij.execution.configurations.RuntimeConfigurationError
+import com.intellij.execution.configurations.RuntimeConfigurationWarning
+import com.intellij.execution.runners.ExecutionEnvironment
+import com.intellij.openapi.project.Project
+import com.microsoft.azure.hdinsight.sdk.common.azure.serverless.AzureSparkServerlessCluster
+import com.microsoft.azure.hdinsight.sdk.common.azure.serverless.AzureSparkServerlessClusterManager
+import com.microsoft.azure.hdinsight.sdk.common.livy.interactive.ServerlessSparkSession
+import com.microsoft.azure.hdinsight.spark.common.ServerlessSparkSubmitModel
+import com.microsoft.azure.hdinsight.spark.run.configuration.RemoteDebugRunConfiguration
+import java.net.URI
+
+class ServerlessSparkScalaLivyConsoleRunConfiguration(project: Project,
+                                            configurationFactory: SparkScalaLivyConsoleRunConfigurationFactory,
+                                            batchRunConfiguration: RemoteDebugRunConfiguration?,
+                                            name: String)
+    : SparkScalaLivyConsoleRunConfiguration(
+        project, configurationFactory, batchRunConfiguration, name)
+{
+    private var submitModel = batchRunConfiguration?.submitModel as ServerlessSparkSubmitModel
+
+    private var tenantId = this.submitModel?.tenantId;
+
+    private var accountName = (batchRunConfiguration?.submitModel as ServerlessSparkSubmitModel)?.accountName;
+
+    override var clusterName = batchRunConfiguration?.submitModel?.submissionParameter?.clusterName
+            ?: throw RuntimeConfigurationWarning("An Azure Data Lake Spark Run Configuration should be selected to start a console")
+
+    override fun getState(executor: Executor, env: ExecutionEnvironment): RunProfileState? {
+        val session = ServerlessSparkSession(
+                name,
+                URI.create((cluster as? AzureSparkServerlessCluster)?.livyUri.toString()),
+                accountName,
+                tenantId
+        )
+
+        return SparkScalaLivyConsoleRunProfileState(SparkScalaConsoleBuilder(project), session)
+    }
+
+    override fun checkSettingsBeforeRun() {
+        cluster = AzureSparkServerlessClusterManager
+                .getInstance()
+                .clusters
+                .find { it.name == clusterName }?:throw RuntimeConfigurationError("Can't find the target cluster $clusterName");
```

Comment:
Break the line before `?:` might be a good practise.

Reply:
Done.